### PR TITLE
add unit tests for getStorageAccounts in azure_storageaccount.go

### LINKED ISSUE #94240: write unit tests for azure_storageaccount.go
**What would you like to be added**:

`getStorageAccounts` and `EnsureStorageAccount` functions need unit tests
https://github.com/kubernetes/kubernetes/blob/8c9d4def18019f3b39305fa83eddedd9f2bc930a/staging/src/k8s.io/legacy-cloud-providers/azure/azure_storageaccount.go#L43

**Why is this needed**:

/kind test
/sig cloud-provider
/area provider/azure

/assign

### PR DESCRIPTION
**Pull request by antcs**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
add unit tests for `getStorageAccounts` in `azure_storageaccount.go`

See: https://github.com/kubernetes/kubernetes/blob/8c9d4def18019f3b39305fa83eddedd9f2bc930a/staging/src/k8s.io/legacy-cloud-providers/azure/azure_storageaccount.go#L43

**Which issue(s) this PR fixes**:
Partially Fixes #94240

**Special notes for your reviewer**:
First PR from my side

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_storageaccount_test.go [hunk 1]
```diff
@@ -84,3 +84,171 @@ func TestGetStorageAccessKeys(t *testing.T) {
...
+
+	cloud := &Cloud{}
+
+	name := "testAccount"
+	location := "testLocation"
+	networkID := "networkID"
+	accountProperties := storage.AccountProperties{
+		NetworkRuleSet: &storage.NetworkRuleSet{
+			VirtualNetworkRules: &[]storage.VirtualNetworkRule{
+				{VirtualNetworkResourceID: &networkID,
```
Comment (nilo19) [inline staging/src/k8s.io/legacy-cloud-providers/azure/azure_storageaccount_test.go:100]: nit: please leave `{` and `}` in one separated line

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_storageaccount_test.go [hunk 2]
```diff
@@ -84,3 +84,171 @@ func TestGetStorageAccessKeys(t *testing.T) {
...
+	}}
+
+	tests := []struct {
+		testCase           string
+		testAccountOptions *AccountOptions
+		testResourceGroups []storage.Account
+		expectedResult     []accountWithLocation
+		expectedError      error
+	}{
+		{testCase: "account name is nil",
```
Comment (nilo19) [inline staging/src/k8s.io/legacy-cloud-providers/azure/azure_storageaccount_test.go:194]: same here, place them in one separated line

Comment (antcs) [reply staging/src/k8s.io/legacy-cloud-providers/azure/azure_storageaccount_test.go]: done

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_storageaccount_test.go (hunk 1 above)
Comment (antcs) [reply staging/src/k8s.io/legacy-cloud-providers/azure/azure_storageaccount_test.go]: done

**Comment by antcs** [comment]
@andyzhangx @nilo19 review addressed and the pipelines are all green

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve
thanks for the contribution!

**Pull request merged**